server/routerlicious/packages/lambdas-driver/src/kafka-service/checkpointManager.ts

```diff
@@ -65,7 +65,11 @@ export class CheckpointManager {
 
 				// Trigger another checkpoint round if the offset has moved since the checkpoint finished and
 				// resolve any pending checkpoints to it.
-				if (this.lastCheckpoint && this.lastCheckpoint !== this.commitedCheckpoint) {
+				if (
+					this.lastCheckpoint &&
+					(this.lastCheckpoint.partition !== this.commitedCheckpoint.partition ||
```

Comment:
I'm curious about the partition id comparison. Do you know if that has ever actually happened? 
The session (tenant/document id pair) should be bound to a specific partition. It's concerning if it randomly changed for a running deli lambda.

Reply:
That makes sense, I havent seen that happening for deli/scribe but just added that check in case that happens for other lambdas. For example, for non-document lambdas like scriptorium, I believe a single lambda might be processing multiple partitions, right? Please correct me if I am wrong or misunderstanding.

Reply:
There should be 1 instance of this class per [Partition](https://github.com/microsoft/FluidFramework/blob/25cd6a9cd7e084415f7ec4a0ae5e178795e0a650/server/routerlicious/packages/lambdas-driver/src/kafka-service/partition.ts#L18), and the constructor the CheckpointManager class has the partition id in it. It might be worth adding an assert to verify `this.id === queuedMessage.partition`

Reply:
There's also an assert at the start of this method `this.lastCheckpoint === undefined || queuedMessage.offset >= this.lastCheckpoint.offset,` - it looks like it already ensures the checkpoint keeps moving forward?

Reply:
Got it, thanks for clarifying about the partition.
And yeah you are right, we do have a check to ensure that checkpoint keeps moving forward at the beginning of this method, so this change shouldnt be needed. I somehow thought that its possible to get an earlier offset here, but I will investigate a bit more on how is that happening. :)